### PR TITLE
feat: adds relative-ci agent

### PR DESCRIPTION
**Pull request by dzucconi**
Closes [DIA-793](https://artsyproduct.atlassian.net/browse/DIA-793)

We need to generate size diffs for bundle reports on pull requests. Testing out [relative-ci.com](https://relative-ci.com/) — free for open source.

[DIA-793]: https://artsyproduct.atlassian.net/browse/DIA-793?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by egdbear — DISMISSED** [review]
🚀

**Review by araujobarret — COMMENTED** [review]
Can't we generate the artifacts to be stored within the job?
Might be nicer to take than the console

**Comment by dzucconi** [comment]
@araujobarret I don't really follow what you're asking?

**Comment by araujobarret** [comment]
image

Does this step generate something more visual to take a look(Artifacts in this case)?

**Comment by dzucconi** [comment]
Still not sure what you're asking. I'm using an external service. It will indeed generate something visual.

**Comment by araujobarret** [comment]
That was my question, since it's not working yet and there's no screenshot in the PR I wanted to see where something more visual would come in(didn't look at the tool so much to say, maybe I should)

**Comment by dzucconi** [comment]
Yeah, lemme get this working first 😄

**Comment by damassi** [comment]
Ahh nice output, and not in house 👌

**Comment by araujobarret** [comment]
> Yeah, lemme get this working first 😄

a'ight 😂 

is there a way to prevent generated files from being tagged as duplicated?

**Review by araujobarret — DISMISSED** [review]

**Review by damassi — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -330,6 +341,12 @@ workflows:
           requires:
             - electron-runner-image-push
 
+      - bundle-stats:
+          <<: *not_staging_or_release
+          context: hokusai
```
Comment (joeyAghion) [inline .circleci/config.yml:346]: What permissions are actually necessary for this step? This context has access to release to (and read from) our cluster which I imagine isn't necessary. If access to S3 is necessary, there's probably a more restrictive context we can lean on.

Comment (dzucconi) [reply .circleci/config.yml:346]: Right, this just needs to run the build to emit the stats then upload them using the agent + the environment variable that's set in Circle. I, however, don't understand the config or how to use a more restrictive context, in this case.

File: Dockerfile [hunk 2]
```diff
@@ -118,6 +119,7 @@ COPY .env.oss \
   jest.config.js \
   package.json \
   relay.config.js \
+  relativeci.config.js \
```
Comment (starsirius) [inline Dockerfile:122]: I thought we need to update [`.dockerignore`](https://github.com/artsy/force/blob/37dc7545141ff936a9ffbf85190fd04c3a8a5e2d/.dockerignore) accordingly, given we explicitly allow individual files. But looks like we actually don't! TIL!

File: .circleci/config.yml (hunk 1 above)
Comment (joeyAghion) [reply .circleci/config.yml:346]: > upload them using the agent + the environment variable that's set in Circle

Which environment variable?

Comment (dzucconi) [reply .circleci/config.yml:346]: `RELATIVE_CI_KEY`

Comment (dzucconi) [reply .circleci/config.yml:346]: So, what context should I use though? I apparently need the AWS creds just to run the build.

Comment (joeyAghion) [reply .circleci/config.yml:346]: If you mean the prerequisite builder-image-push step by "the build," I think that already uses the `hokusai` context. Does this step also need AWS creds? If not, I'd create a new `relativeci` org context, add the key there, and specify it here.

Comment (dzucconi) [reply .circleci/config.yml]: Just did that. Apparently not:



In the interest of moving on I'm going to proceed with the Hokusai context for now.

**Pull request merged**